i18n-helpers/src/lib.rs

```diff
@@ -333,18 +335,140 @@ fn is_comment_skip_directive(html: &str) -> bool {
     re.is_match(html.trim())
 }
 
-/// Returns true if the events appear to be a codeblock without translatable text.
-fn is_nontranslatable_codeblock_group(events: &[(usize, Event)]) -> bool {
-    match events {
-        [(_, Event::Start(Tag::CodeBlock(_))), .., (_, Event::End(Tag::CodeBlock(_)))] => {
-            let (codeblock_text, _) = reconstruct_markdown(events, None);
-            // Heuristic to check whether the codeblock nether has a
-            // literal string nor a line comment.  We may actually
-            // want to use a lexer here to make this more robust.
-            !codeblock_text.contains('"') && !codeblock_text.contains("//")
+/// Returns true if the events appear to be a codeblock.
+fn is_codeblock_group(events: &[(usize, Event)]) -> bool {
+    matches!(
+        events,
+        [
+            (_, Event::Start(Tag::CodeBlock(_))),
+            ..,
+            (_, Event::End(Tag::CodeBlock(_)))
+        ]
+    )
+}
+
+/// Returns true if the scope should be translated.
+fn is_translate_scope(x: Scope) -> bool {
+    let scope_string = Scope::new("string").unwrap();
+    let scope_comment = Scope::new("comment").unwrap();
+    scope_string.is_prefix_of(x) | scope_comment.is_prefix_of(x)
```

Comment:
I believe it should be fine to only evaluate one side of the `OR` with a Boolean short-circuiting operator:

```suggestion
    scope_string.is_prefix_of(x) || scope_comment.is_prefix_of(x)
```